docs/DeveloperGuide.md

```diff
@@ -1,34 +1,662 @@
+
+
+### 3.4. Event
+The diagram below shows the architecture for Event feature. (Coming soon)
+
+![](EventDiagram/EventArchitecture.png)
+
+
+There are a total of 6 commands under Event feature.
+ `CommandEventAdd`, `CommandEventDel`, `CommandEventList`  ,`CommandEventStatus`, `CommandSearchEvent` , `CommandEventCountdown` , `CommandAddEventAttendance`(coming soon),`CommandDelEventAttendance`(coming soon), `CommandViewEventAttendence`(coming soon). 
+ 
+ The implementation for each command is described in detail below.
+                                                             
+**3.4.1. Add/delete events feature** `CommandEventAdd` , `CommandEventDel` 
+
+**3.4.1.1. Current Implementation**
+The `CommandEventAdd` class in `seedu.duke.event` handles the adding of events. According to the `userInput`, it adds a new event to the `EventList`. 
+The `CommandEventDel` class in the same package handles deleting of a event. It deletes an `Event` instance according to the index provided by `userInput` from the `EventList`.  
+They implement the following operations:  
+* `CommandEventAdd#execute()` - Adds a new event into the `EventList` according to `userInput`.  
+* `CommandEventDel#execute()` - Deletes an event from `EventList` or deletes all the events in the list. 
+To delete a particular event, enter the index of the event.
+To delete all the events in the list, enter `all` instead of the index of the event.
+
+Given below is an example usage scenario and how add/delete event function behaves at each step.  
+
+Step 1. The user launches the application for the first time.   
+
+![](EventDiagram/Step1.png)
+
+Step 2. The user executes `event addEvent /n arduino course /d 2020-12-30 /t 8pm` command to add a new event with the name "arduino course", 
+the date of the event "2020-12-30" and the time "8pm" into event list. 
+The `event addEvent` command calls `CommandEventAdd#execute()`, then `EventList` will add a new `Event` with event name as `iphone12`, date as `2020-12030` and time as `8pm`.  
+
+![](EventDiagram/Step2.png)
+
+Step 3. The user executes `event delEvent 1` command to delete the 1st event in the event list. The `event delEvent`
+command calls `CommandEventDel#execute()`, causing the `Event` at index 1 to be removed from `EventList`.  
+
+![](EventDiagram/Step3.png)
+
+The sequence diagram for adding an event is as shown below:
+
+![CommandEventAdd](EventDiagram/SequenceDiagram/CommandEventAdd.png)
+
+**3.4.1.2. Design Considerations**
+Aspect : User adds the same event multiple times
+Alternative(current choice) : The program will remind user that the event has already been added.
+
+Aspect : User adds an event with a past date
+Alternative(current choice) : The program will remind user that the date is past.
+
+The sequence diagram for deleting a particular event or all events is as shown below:
+
+![CommandEventDelete](EventDiagram/SequenceDiagram/CommandEventDelete.png)
```

Comment:
<img width="530" alt="Screenshot 2020-10-30 at 12 20 59" src="https://user-images.githubusercontent.com/59906930/97659710-9fa3b280-1aaa-11eb-9ea8-d2b7e266bc59.png">

Should the dotted lines stop after the deletion of "Event"? (after the X)